tests/meta/clean/test_clean_demultiplexed_runs.py

```diff
@@ -0,0 +1,99 @@
+"""Tests for cleaning demultiplexed-runs using
+cg.meta.clean.demultiplexed_flowcells.DemuxedFlowcell """
+from pathlib import Path
+from typing import Optional
+from unittest import mock
+
+import pytest
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.meta.clean.demultiplexed_flowcells import DemultiplexedRunsFlowcell
+from cg.store import Store
+from cg.store.models import Flowcell
+
+
+@pytest.mark.parametrize(
+    "flowcell_path, result",
+    [
+        ("correct_flowcell_path", True),
+        ("incorrect_flowcell_path_too_long", False),
+        ("incorrect_flowcell_path_extension", False),
+    ],
+)
+@mock.patch("cg.apps.housekeeper.hk.HousekeeperAPI")
+@mock.patch("cg.store.Store")
+def test_flowcell_name(
+    mock_statusdb: Store, mock_hk: HousekeeperAPI, flowcell_path: Path, result: bool, request
+):
+    # GIVEN a flowcell
+    flowcell_path = request.getfixturevalue(flowcell_path)
+    mock_hk.files.return_value.count.return_value = 1
```

Comment:
There's a danger that this deep mocking hinders changes to the implementation, see https://www.youtube.com/watch?v=qPfQM4w4I04